components/org.wso2.carbon.identity.application.authenticator.push/src/main/java/org/wso2/carbon/identity/application/authenticator/push/notification/handler/FirebasePushNotificationSender.java

```diff
@@ -82,7 +82,8 @@ public static synchronized FirebasePushNotificationSender getInstance() {
      */
     public void sendPushNotification(String deviceId, String pushId, String message, String randomChallenge,
                                      String sessionDataKey, String username, String fullName, String organization,
-                                     String serviceProviderName, String hostname, String userOS, String userBrowser)
+                                     String serviceProviderName, String hostname, String userOS, String userBrowser,
+                                     String metadata)
```

Comment:
update java doc comment

Reply:
Also whats passed in as metadata? looking at the param name should it be a string?

Reply:
This method seems to be having a lot of arguments. Let's create a model class and pass them in from that.